contracts/src/v0.8/feeds/DualAggregator.sol

```diff
@@ -524,17 +546,17 @@ contract DualAggregator is OCR2Abstract, OwnerIsCreator, AggregatorV2V3Interface
 
     // decreasing loop from the latest primary round id
     for (uint32 round_ = latestAggregatorRoundId; round_ > 0; --round_) {
+      // in case it's the latest secondary round id, return it
+      if (round_ == s_hotVars.latestSecondaryRoundId || latestAggregatorRoundId - round_ == s_maxSyncIterations) {
```

Comment:
Small gas optimization: store s_maxSyncIterations in a variable above the for loop and use that variable instead.

Reply:
Also this isn't correct. If it reaches the max iterations without having found the round, we shouldn't return anything. We should revert.

Reply:
We should also add a test case that covers exceeded the max iterations

Reply:
Not sure about reverting it, this internal function is called for all the `get` and `call` methods.

Reverting it we make them all inaccessible and we can break many client applications.

I would simply have it to stop the loop, just like we have the cutoff time or the comparison with the `latestSecondaryRoundId`.

Reply:
Also, I'm not sure if it's really useful, since we already have the cutoff time to stop the loop, the maxIterations should always be bigger than the cutoff time, if not we would be returning a newer price than the cutoff time.

The only two cases I see:
1. cutoffTime > maxIterations -> maxIterations has no effect to the loop.
2. cutoffTime < maxIterations -> if we need the maxIterations, this means that the cutoffTime is too big and we should reduce it (also, we would be returning a newer round than the cutoff time).

Reply:
It's a valid point. The only purpose of the maxIterations variable would be so that we don't ever run into an out of gas error. As you mentioned, this would only happen if we have a very large cutoff time, and the secondary feed has been down for a long time.

But we should still have defined behaviour for those cases.

I just realized we have 2 helper functions that loop through rounds:

1.  _doesReportExist (called when writing)
2. _getSyncPrimaryRound (called when reading)

For reading, if we reach maxIterations then returning latestSecondaryRoundId is the behaviour we want.

For writing, I believe if we reach maxIterations we should revert.